### ISSUE TITLE
Compilation fails when using gulp

### ISSUE THREAD

**Issue by andy-ms** [Microsoft/TypeScript#17500]
`gulp clean` and `gulp runtests`.
Get:
```
src/services/jsTyping.ts(160,37): error TS2453: The type argument for type parameter 'T' cannot be inferred from the usage. Consider specifying the type arguments explicitly.
  Type argument candidate 'MapLike<string>' is not a valid type argument because it is not a supertype of candidate 'string'.
src/services/jsTyping.ts(160,37): error TS2453: The type argument for type parameter 'T' cannot be inferred from the usage. Consider specifying the type arguments explicitly.
  Type argument candidate 'MapLike<string>' is not a valid type argument because it is not a supertype of candidate 'string'.
```

Is gulp doing something different from jake?

**Comment by weswigham** [comment]
Gulp uses our services to build instead of our command line compiler. Other than that... it runs through `gulp-typescript` before it gets to us?

**Comment by weswigham** [comment]
@andy-ms Our services layer under `jake` is being built with the `master` compiler, while under `gulp`, it must be built with our `LKG` compiler. An LKG update should fix this.

**Comment by weswigham** [comment]
This should be fixed with #17538.
